addons/portal/static/src/js/components/input_confirmation_dialog/input_confirmation_dialog.js

```diff
@@ -0,0 +1,47 @@
+/** @odoo-module */
+
+import { useEffect } from "@odoo/owl";
+import { ConfirmationDialog } from "@web/core/confirmation_dialog/confirmation_dialog";
+
+export class InputConfirmationDialog extends ConfirmationDialog {
+    static props = {
+        ...ConfirmationDialog.props,
+        onInput: Function,
+    };
+    static template = "portal.InputConfirmationDialog";
+
+    setup() {
+        super.setup();
+
+        const onInput = () => {
+            if (this.props.onInput) {
+                this.props.onInput(this.context);
+            }
+        };
+        const onKeydown = (ev) => {
+            if (ev.key && ev.key.toLowerCase() === "enter") {
+                ev.preventDefault();
+                this._confirm();
+            }
+        };
+        useEffect(
+            (modalEl) => {
+                this.inputEl = modalEl.querySelector("input");
+                if (this.inputEl) {
+                    this.inputEl.focus();
+                    this.inputEl.addEventListener("keydown", onKeydown);
+                    this.inputEl.addEventListener("input", onInput);
+                    return () => {
+                        this.inputEl.removeEventListener("keydown", onKeydown);
+                        this.inputEl.removeEventListener("input", onInput);
+                    };
+                }
+            },
+            () => [this.modalRef.el]
```

Comment:
should depend directly on the inputEl: `this.modalRef.el?.querySelector("input")`